drivers/tty/serial/8250/8250.h

```diff
@@ -49,6 +49,9 @@ struct uart_8250_dma {
 	unsigned char		tx_running;
```

Comment:
This patch is a re-write of earlier "serial: 8250_dma: use linear buffer for transmit". That had the following log message:

> 8250_dma used the circular xmit->buf as DMA output buffer. This causes messages that wrap around in the circular buffer to be transmitted using 2 DMA transfers. Depending on baud rate and processor load this can cause an interchar gap in the middle of the message. On the receiving end the gap may cause a short receive timeout, possibly long enough to terminate a DMA transfer, but too short to restart a receive DMA transfer in time thus causing a receive buffer overrun. Fix this but creating a linear tx_buffer and copying all of xmit->buf into it.

I will use an appropriate modified text for this patch after you comments.